designs/multiple-worker-node-groups.md

```diff
@@ -0,0 +1,1273 @@
+# Multiple worker node groups
+
+## Introduction
+
+**Problem:** Currently users of eks-anywhere can only create one worker node group by specifying the configuration details in a cluster spec.
+
+This limits the ability of a user to create worker nodes with different configurations and formulate application deployment strategy based on the configurations of the worker nodes.
+
+Also, this creates a problem while tainting a node. Unless, eks-anywhere clusters support multiple worker node groups, worker nodes can not be tainted with 'NoExecute' effect, since nodes in a node group have the same configuration, setting a taint with 'NoExecute' effect would essentially make the worker nodes unusable for eks-anywhere specific deployments, as the general approach for eks-anywhere is to not add tolerations on deployments.  
```

Comment:
```suggestion
Also, this creates a problem while tainting a node. Unless, eks-anywhere clusters support multiple worker node groups, worker nodes can not be tainted with `NoExecute` effect, since nodes in a node group have the same configuration, setting a taint with `NoExecute` effect would essentially make the worker nodes unusable for eks-anywhere specific deployments, as the general approach for eks-anywhere is to not add tolerations on deployments.  
```